cql3/functions/functions.cc

```diff
@@ -159,6 +160,20 @@ std::optional<function_name> functions::used_by_user_aggregate(const function_na
     return {};
 }
 
+std::optional<function_name> functions::used_by_user_function(const ut_name& user_type) {
+    for (const shared_ptr<function>& fptr : _declared | boost::adaptors::map_values) {
+        for (auto& arg_type : fptr->arg_types()) {
```

Comment:
Nitpick: shouldn't we just immediately skip functions from the wrong keyspace instead of making all the following tests that won't succeed because the keyspace doesn't match?

Reply:
A function from one keyspace can use types from another.

Reply:
Actually, is that true @avikivity? In `cql3_type::raw_ut` we have https://github.com/scylladb/scylladb/blob/ccc8e47db1807f3cd5088b3cebf0627c1c28dcff/cql3/cql3_type.cc#L226-L231 and this error is used when I try to create a function that uses a type from another keyspace.
Or maybe current behavior is a bug